File: Euro.java
public class Euro extends Currency {
    private static final double RATE = 42.0;

    public Euro(double amount) {
        super(amount);
    }

    @Override
    public double toUAH() {
        return amount * RATE;
    }

    @Override
    public void print() {
        System.out.println("Euro: " + amount + " => " + toUAH() + " UAH");
    }
}

File: lab8/lab8/Euro.cs
using System;

namespace OOP_Lab9_Variant19
{
    public class Euro : Currency
    {
        private const double rate = 42.0; // 1 євро ≈ 42 грн

        public Euro(double amount) : base(amount) { }

        public override double ToUAH()
        {
            return amount * rate;
        }

        public override void Print()
        {
            Console.WriteLine($"Euro: {amount} => {ToUAH()} UAH");
        }
    }
}